pkg/clusterconditions/cache/cache.go

```diff
@@ -0,0 +1,169 @@
+// Package cache implements a throttled, caching condition.
```

Comment:
> After evaluating a PromQL query, the cluster-version operator will
>  wait at least 10 minutes before evaluating any PromQL

So a given `Condition` will take a minimum of (#PromQL queries * 10) minutes to evaluate?  Am I understanding this correctly?

Reply:
assuming all the queries are net-new and not something that was already in the cache, yes.  Not very snappy.  On the other hand, Thanos load will be really low.  And in the meantime, the conditional edges will be `Recommended=Unknown`, which isn't the end of the world, because there are probably not all that many cases where it's important to be `Recommended=True` with low latency.